examples/cryptocurrency-advanced/backend/src/config.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright 2020 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Cryptocurrency configuration.
+
+use exonum_derive::{BinaryValue, ExecutionFail};
+use exonum_proto::ProtobufConvert;
+
+use super::proto;
+
+/// Cryptocurrency configuration parameters.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[derive(ProtobufConvert, BinaryValue)]
+#[protobuf_convert(source = "proto::Config")]
+pub struct Config {
+    /// Initial balance of the wallet.
+    pub init_balance: i64,
+}
+
+impl Config {
+    /// Verifies that initial balance is more then zero.
+    pub fn verify(&self) -> Result<(), ConfigError> {
+        if self.init_balance < 0 {
+            Err(ConfigError::NegativeBalance)
+        } else {
+            Ok(())
+        }
+    }
+}
+
+/// The enumeration represents errors that can occurred while validating config.
+#[derive(Debug, ExecutionFail)]
+pub enum ConfigError {
+    /// Initial balance is negative.
+    NegativeBalance = 1,
```

Comment:
The code of this error will clash with the `Error::SenderNotFound` variant defined in the `transactions` module. It could make sense to start enumeration here from 16 or 32 (and, ideally, explain in a comment why such separation is beneficial for forward compatibility).

Reply:
Note that such a separation is already introduced in [supervisor](https://github.com/exonum/exonum/blob/master/services/supervisor/src/errors.rs).